armotypes/collaborationconfig.go

```diff
@@ -44,13 +44,22 @@ type CollaborationConfigOption struct {
 	IconBase64 string `json:"iconBase64,omitempty" bson:"iconBase64,omitempty"`
 }
 
+type ChannelProvider string
+
+const (
+	CollaborationTypeJira  ChannelProvider = "jira"
+	CollaborationTypeSlack ChannelProvider = "slack"
+	CollaborationTypeTeams ChannelProvider = "teams"
+	CollaborationTypeEmail ChannelProvider = "email"
+)
+
 // swagger:model CollaborationConfig
 type CollaborationConfig struct {
 	PortalBase `json:",inline" bson:",inline"`
 
 	// Provider name
 	// Example: jira
-	Provider string `json:"provider" bson:"provider,omitempty"`
+	Provider ChannelProvider `json:"provider" bson:"provider,omitempty"`
```

Comment:
comment to exlain why both can be omitempty or not...

Reply:
add omit empty to json too